bigchaindb/core.py

```diff
@@ -174,36 +174,38 @@ def get_tx_by_payload_hash(self, payload_hash):
             returns `None`
         """
 
-        cursor = r.table('bigchain')\
-            .get_all(payload_hash, index='payload_hash')\
+        cursor = r.table('bigchain') \
+            .get_all(payload_hash, index='payload_hash') \
             .run(self.conn)
 
         transactions = list(cursor)
         return transactions
 
-    def get_spent(self, txid):
+    def get_spent(self, tx_input):
         """Check if a `txid` was already used as an input.
 
         A transaction can be used as an input for another transaction. Bigchain needs to make sure that a
         given `txid` is only used once.
 
         Args:
-            txid (str): transaction id.
+            tx_input (dict): Input of a transaction in the form `{'txid': 'transaction id', 'cid': 'condition id'}`
 
         Returns:
             The transaction that used the `txid` as an input if it exists else it returns `None`
         """
         # checks if an input was already spent
-        # checks if the bigchain has any transaction with input `transaction_id`
+        # checks if the bigchain has any transaction with input {'txid': ..., 'cid': ...}
         response = r.table('bigchain').concat_map(lambda doc: doc['block']['transactions'])\
-            .filter(lambda transaction: transaction['transaction']['input'] == txid).run(self.conn)
+            .filter(lambda transaction: transaction['transaction']['fulfillments']
+                    .contains(lambda fulfillment: fulfillment['input'] == tx_input))\
+            .run(self.conn)
 
         # a transaction_id should have been spent at most one time
         transactions = list(response)
         if transactions:
             if len(transactions) != 1:
                 raise Exception('`{}` was spent more then once. There is a problem with the chain'.format(
```

Comment:
Do not use generic `Exception`s. There is a `bigchaindb.exceptions.DoubleSpent`. If it's not suitable for this case, create a new one.